cmd/habitat/community/consensus/raft/service.go

```diff
@@ -0,0 +1,289 @@
+package raft
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/eagraf/habitat/pkg/raft/fsm"
+	"github.com/eagraf/habitat/pkg/raft/transport"
+	"github.com/hashicorp/raft"
+	raftboltdb "github.com/hashicorp/raft-boltdb"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	RetainSnapshotCount = 1000
+	RaftTimeout         = 10 * time.Second
+)
+
+// ClusterService is an implementation of cluster.ClusterManager
+type ClusterService struct {
+	mux       transport.RaftMultiplexer
+	instances map[string]*raftClusterInstance
+
+	nodeID string
+}
+
+type raftClusterInstance struct {
+	communityID  string
+	serverID     string
+	address      string
+	instance     *raft.Raft
+	stateMachine *fsm.CommunityStateMachine
+}
+
+func NewClusterService() *ClusterService {
+	cs := &ClusterService{
+		mux:       *transport.NewRaftMultiplexer(),
+		instances: make(map[string]*raftClusterInstance),
+
+		nodeID: compass.NodeID(),
+	}
+
+	return cs
+}
+
+func (cs *ClusterService) Start() error {
+	go cs.mux.Listen(getMultiplexerAddress())
```

Comment:
does this need to defer close it somewhere?

Reply:
Since theres no way to modify the cluster services being used (just raft for now), I don't think we need to worry about adding this until later